internal/storage/fs/snapshot.go

```diff
@@ -0,0 +1,86 @@
+package fs
+
+import (
+	iofs "io/fs"
+
+	"github.com/gobwas/glob"
+	"go.uber.org/zap"
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	indexFile = ".flipt.yml"
+)
+
+var (
+	// globDefaults will be used if index file does not exist
+	// or can not be successfully read (malformed).
+	globDefaults []string = []string{"**/features.yml", "**/features.yaml", "**/*.features.yml", "**/*.features.yaml"}
+)
+
+// FliptIndex represents the structure of a well-known file ".flipt.yml"
+// at the root of an FS.
+type FliptIndex struct {
+	Version string   `yaml:"version,omitempty"`
+	Include []string `yaml:"include,omitempty"`
+	Exclude []string `yaml:"exclude,omitempty"`
+}
+
+// Snapshot represents the flag state from a FS.
+type Snapshot struct{}
+
+func buildSnapshotHelper(logger *zap.Logger, source iofs.FS) ([]string, []glob.Glob, error) {
```

Comment:
I created this helper function for easier testing. I wanted to assert against the lengths of the slices that were being returned from the function.